nagra/transaction.py

```diff
@@ -1,16 +1,83 @@
 import sqlite3
 import threading
+from typing import Callable
 
-from nagra.utils import logger
+from nagra.utils import logger, UNSET
 from nagra.exceptions import NoActiveTransaction
 
 
+class LRUGenerator:
+    """
+    LRUGenerator wraps generator calls and implement LRU caching
+    """
+
+    def __init__(self, generator, size=1000):
+        self.size = size
+        self.generator = generator
+        self.recent = {}
+        self.older = {}
+
+    def run(self, keys):
+        # Capture existing known values in `cached`
+        cached = {}
+        fresh = []
+        for k in keys:
+            v = self.get(k)
```

Comment:
@bertrandchenal I understand how it works, but I don't understand on what you call the `get` method here.